src/v/pandaproxy/client/test/fetch.cc

```diff
@@ -0,0 +1,112 @@
+    auto ntp = make_default_ntp(tp.topic, tp.partition);
+
+    auto log_config = make_default_config();
+    {
+        using namespace storage;
+        storage::disk_log_builder builder(log_config);
+        storage::ntp_config ntp_cfg(
+          ntp, log_config.base_dir, nullptr, storage::ntp_config::ntp_id(2));
+        builder | start(std::move(ntp_cfg)) | add_segment(model::offset(0))
+          | add_random_batch(model::offset(0), 10, maybe_compress_batches::yes)
+          | stop();
+    }
+
+    info("Waiting for leadership");
+    wait_for_controller_leadership().get();
+
+    info("Connecting client");
+    ppc::shard_local_cfg().retry_base_backoff.set_value(10ms);
+    ppc::shard_local_cfg().retries.set_value(size_t(1));
+    auto client = make_connected_client();
+    client.connect().get();
+
+    {
+        info("Fetching from unknown topic");
+        auto p{
+          client.fetch_partition(tp, model::offset(0), 1024, 1000ms).get()};
+        BOOST_REQUIRE_EQUAL(p.name, tp.topic);
+        BOOST_REQUIRE_EQUAL(p.responses.size(), 1);
+        BOOST_REQUIRE_EQUAL(p.responses[0].id, tp.partition);
+        BOOST_REQUIRE_EQUAL(
+          p.responses[0].error, kafka::error_code::unknown_topic_or_partition);
+    }
+
+    info("Adding known topic");
+    add_topic(model::topic_namespace_view(ntp)).get();
+    auto shard = app.shard_table.local().shard_for(ntp);
+    {
+        info("Waiting for topic leader");
+        tests::cooperative_spin_wait_with_timeout(10s, [this, shard, ntp] {
+            return app.partition_manager.invoke_on(
+              *shard, [ntp](cluster::partition_manager& mgr) {
+                  auto partition = mgr.get(ntp);
+                  return partition && partition->is_leader();
+              });
+        }).get();
+    }
+
+    model::offset committed_offset{};
+    info("Waiting for topic data");
+    tests::cooperative_spin_wait_with_timeout(
+      10s,
+      [this, shard, ntp, &committed_offset] {
+          return app.partition_manager.invoke_on(
+            *shard, [ntp, &committed_offset](cluster::partition_manager& mgr) {
+                auto partition = mgr.get(ntp);
+                if (partition) {
+                    committed_offset = partition->committed_offset();
+                    return committed_offset >= model::offset(1);
+                }
+                return false;
+            });
+      })
+      .get();
+
+    {
+        info("Fetching from nonempty known topic");
+        ppc::shard_local_cfg().retries.set_value(size_t(3));
+        auto p{
+          client.fetch_partition(tp, model::offset(0), 1024, 1000ms).get()};
+        BOOST_REQUIRE_EQUAL(p.name, tp.topic);
+        BOOST_REQUIRE_EQUAL(p.responses.size(), 1);
+        auto const& r = p.responses[0];
+        BOOST_REQUIRE_EQUAL(r.id, tp.partition);
+        BOOST_REQUIRE_EQUAL(r.error, kafka::error_code::none);
+        BOOST_REQUIRE_EQUAL(r.high_watermark, committed_offset);
```

Comment:
Maybe there'd be some value in verifying the returned data against the expected

Reply:
Perhaps. Any thoughts on how to populate the log with a higher level API than the `disk_log_builder | add_random_batch`? I could just use the proxy produce API, but that feels too high level, and makes the tests at this layer of abstraction a bit too reliant on each other.

Reply:
For copro i created this method: https://github.com/vectorizedio/redpanda/blob/dev/src/v/coproc/tests/coproc_test_fixture.cc#L135

The method probably would be a better fit as a member of rp_thread_fixture anyway